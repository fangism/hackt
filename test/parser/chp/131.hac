// "131.hac"
// accepts-invalid bug

defproc foo(int sel; chan?(bool) in; chan!(bool) out) {
bool x;

chp {
x := true;
*[
[ x ->
	in[sel]?(x);	// should reject invalid!
	out!(x)
[] else  -> skip
]]
}
}

foo bar;
