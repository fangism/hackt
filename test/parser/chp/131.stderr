ERROR: instance collection in is 0-dimensional, and thus, cannot be indexed 1-dimensionally!  [11:5..7]
ERROR resolving instance reference at [11:2..7]
ERROR in CHP statement(s) at [11:2--12:8]
